tests/unit/onyxTest.js

```diff
@@ -630,4 +631,95 @@ describe('Onyx', () => {
                 expect(mockCallback.mock.calls[1][0]).toEqual(collectionUpdate);
             });
     });
+
+    it('should not update a subscriber if the value in the cache has not changed at all', () => {
+        const mockCallback = jest.fn();
+        const collectionUpdate = {
+            test_policy_1: {ID: 234, value: 'one'},
+        };
+
+        // Given an Onyx.connect call with waitForCollectionCallback=true
+        connectionID = Onyx.connect({
+            key: ONYX_KEYS.COLLECTION.TEST_POLICY,
+            waitForCollectionCallback: true,
+            callback: mockCallback,
+        });
+        return waitForPromisesToResolve()
+            .then(() => {
+                // When merge is called with an updated collection
+                Onyx.merge(`${ONYX_KEYS.COLLECTION.TEST_POLICY}${1}`, collectionUpdate.test_policy_1);
+                return waitForPromisesToResolve();
+            })
```

Comment:
```suggestion
            // When merge is called with an updated collection
            .then(() => Onyx.merge(`${ONYX_KEYS.COLLECTION.TEST_POLICY}${1}`, collectionUpdate.test_policy_1))
```
I think this is cleaner since `waitForPromisesToResolve()` doesn't need to be used. Looks like this can be applied to some of the other blocks as well

Reply:
Updated some of these.